doc/concepts/configuration.rst

```diff
@@ -0,0 +1,651 @@
+      routers:
+        app:
+          module: router
+          # ...
+
+Learn more about using Tarantool as the application server from :ref:`Developing applications with Tarantool <how-to-app-server>`.
+
+
+
+.. _configuration_predefined_variables:
+
+Predefined variables
+********************
+
+In a configuration file, you can use the following predefined variables that are replaced with actual values at runtime:
+
+-   ``instance_name``
+-   ``replicaset_name``
+-   ``group_name``
+
+To reference such variables in a configuration file, use double curly braces.
+In the example below, ``{{ instance_name }}`` is replaced with *instance001*.
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/templating/config.yaml
+    :language: yaml
+    :dedent:
+
+As a result, the :ref:`paths to snapshots and write-ahead logs <configuration_options_directories>` differ for different instances.
+
+
+
+..  _configuration_environment_variable:
+
+Environment variables
+~~~~~~~~~~~~~~~~~~~~~
+
+For each configuration parameter, Tarantool provides two sets of predefined environment variables:
+
+*   Variables whose names start with ``TT_`` are used to substitute parameters specified in a configuration file.
+    This means that these variables have a higher :ref:`priority <configuration_precedence>` than the options specified in a configuration file.
+
+*   Variables whose names start with ``TT_`` and end with ``_DEFAULT`` are used to specify default values for parameters missing in a configuration file.
+    These variables have a lower :ref:`priority <configuration_precedence>` than the options specified in a configuration file.
+
+For example, ``TT_IPROTO_LISTEN`` and ``TT_IPROTO_LISTEN_DEFAULT`` correspond to the ``iproto.listen`` option.
+``TT_SNAPSHOT_DIR`` and ``TT_SNAPSHOT_DIR_DEFAULT`` correspond to the ``snapshot.dir`` option.
+To see all the supported environment variables, execute the ``tarantool`` command with the ``--help-env-list`` :ref:`option <configuration_command_options>`.
+
+.. code-block:: console
+
+    $ tarantool --help-env-list
+
+Below are a few examples that show how to set environment variables of different types, like *string*, *number*, *array*, or *map*:
+
+*   (*String*) In the example below, ``TT_IPROTO_LISTEN`` is used to specify a :ref:`listening host and port <configuration_options_connection>` values:
+
+    ..  code-block:: console
+
+        $ export TT_IPROTO_LISTEN='127.0.0.1:3311'
+
+    To specify several listening addresses, separate them by a comma without space:
+
+    ..  code-block:: console
+
+        $ export TT_IPROTO_LISTEN='127.0.0.1:3311,127.0.0.1:3312'
+
+*   (*Number*) In this example, ``TT_LOG_LEVEL`` is used to set a logging level to 3 (``CRITICAL``):
+
+    ..  code-block:: console
+
+        $ export TT_LOG_LEVEL=3
+
+*   (*Array*) The examples below show how to set the ``TT_SHARDING_ROLES`` variable that accepts an array value.
+    Arrays can be passed in a *simple* ...
```

Comment:
Not sure unfinished sentence with ellipsis is good here, especially with italic. I couldn't understand right away.
`in a simple or JSON format: <two examples>`.